### PR TITLE
J2735 2020 upgrade

### PR DESCRIPTION
**Pull request by jwillmartin**
# PR Details
## Description
Upgrades include new ASN1 folders containing 2020 version of J2735. Changes were made to many files to allow for use of older and this new standard. These changes are indicated by "#if SAEJ2735_SPEC < 2020". 
This line specifies which standard is to be used by V2X Hub: https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/j2735-2020-upgrade/src/tmx/Asn_J2735/CMakeLists.txt#L44

## Motivation and Context
New versions of the standard are being published. These changes are required to keep up with industry needs. 

## How Has This Been Tested?
Tested all message encoding/decoding with these unit tests: 
https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/j2735-2020-upgrade/src/tmx/TmxUtils/test/J2735MessageTest.cpp
Additional testing was completed on a local instance to ensure messages were generated properly and adhered to the updated standard. Encoded messages were cross-checked with an existing decoder: https://www.marben-products.com/decoder-asn1-automotive/

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/tmx/TmxUtils/test/J2735MessageTest.cpp
Comment (jwillmartin) [inline src/tmx/TmxUtils/test/J2735MessageTest.cpp:604]: Not sure when this was removed. Will add back in.

File: src/tmx/Asn_J2735/src/r2020/.vscode/c_cpp_properties.json
Comment (paulbourelly999) [inline src/tmx/Asn_J2735/src/r2020/.vscode/c_cpp_properties.json:1]: No need to comit vscode stuff

**Comment by paulbourelly999** [comment]
We may want to consider holding off on merging this until we cut a release candidate for VRU. This is a very large change set that spans many pieces of commonly used functionality. I think merging this before VRU integration testing poses substantial risk to the testing

File: src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp
Comment (dan-du-car) [inline src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp:97]: Do we need to support both versions or upgrade to one version?

File: src/tmx/TmxUtils/test/J2735MessageTest.cpp
Comment (jwillmartin) [reply src/tmx/TmxUtils/test/J2735MessageTest.cpp]: Added back.

File: src/tmx/Asn_J2735/src/r2020/ANY_ber.c
Comment (jwillmartin) [reply src/tmx/Asn_J2735/src/r2020/ANY_ber.c]: All 2020 files containing this comment were updated to 2023. Thanks!

File: src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp
Comment (jwillmartin) [reply src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp:97]: I'm not really sure. The previous version had this logic to support both, but you could only use one at a time. Do you think we should just make separate tags for different versions or something else?

Comment (jwillmartin) [reply src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp:97]: I agree. I think we should at least phase out anything pre-2020. Nobody will be using anything older anyway, so it doesn't make sense to maintain support.

Comment (dan-du-car) [reply src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp:97]: I am good with how it is now. Maybe later we can come back with a solution

**Review by dan-du-car — APPROVED** [review]

File: src/v2i-hub/CswPlugin/src/XmlCurveParser.cpp
Comment (kjrush) [inline src/v2i-hub/CswPlugin/src/XmlCurveParser.cpp:293]: Was this a typo in the standard previously "duratonTime"?

**Review by kjrush — COMMENTED** [review]
Man, github does not like PRs of this size. Seems overall good to me.

File: src/v2i-hub/CswPlugin/src/XmlCurveParser.cpp
Comment (jwillmartin) [reply src/v2i-hub/CswPlugin/src/XmlCurveParser.cpp:293]: Ha, yeah it was

**Comment by paulbourelly999** [comment]
@jwillmartin @dan-du-car This can be merged into develop now if it is approved. We have release candidates for VRU so it won't impact our testing. Please resolve merge conflicts before merging.

**Review by dan-du-car — APPROVED** [review]

**Pull request merged**
